legistar/events.py

```diff
@@ -109,64 +118,80 @@ def extractRollCall(self, action_detail_url) :
     def ical(self, ical_text):
         value = icalendar.Calendar.from_ical(ical_text)
         return value
-        
+
 
 
 class LegistarAPIEventScraper(LegistarAPIScraper):
 
     def events(self, since_datetime=None):
+        self._events = self._scrapeWebCalendar()
+        self._scraped_events = {}
+
         if since_datetime:
-            params = {'$filter' : "EventLastModifiedUtc gt datetime'{since_datetime}'".format(since_datetime = since_datetime.isoformat())}
+            # Minutes are often published after an event occurs – without a
+            # corresponding event modification. Query all update fields so later
+            # changes are always caught by our scraper, particularly when
+            # scraping narrower windows of time.
+            update_fields = ('EventLastModifiedUtc',
+                             'EventAgendaLastPublishedUTC',
+                             'EventMinutesLastPublishedUTC')
+
+            since_fmt = " gt datetime'{}'".format(since_datetime.isoformat())
+            since_filter = ' or '.join(field + since_fmt for field in update_fields)
+
+            params = {'$filter' : since_filter}
+
         else:
             params = {}
 
         events_url = self.BASE_URL + '/events/'
 
-        web_results = self._scrapeWebCalendar()
-
         for api_event in self.pages(events_url,
                                     params=params,
                                     item_key="EventId"):
-            start = self.toTime(api_event['EventDate'])
+
             # EventTime may be 'None': this try-except block catches those instances.
             try:
                 start_time = time.strptime(api_event['EventTime'], '%I:%M %p')
+
             except TypeError:
                 continue
+
             else:
+                start = self.toTime(api_event['EventDate'])
                 api_event['start'] = start.replace(hour=start_time.tm_hour,
                                                    minute=start_time.tm_min)
-                api_event['status'] = confirmed_or_passed(api_event['start'])
 
-                key = (api_event['EventBodyName'].strip(),
-                       api_event['start'])
+                api_event['status'] = self._event_status(api_event)
 
-                try:
-                    web_event = web_results[key]
-                    yield api_event, web_event
-                except KeyError:
+                # Skip events that do not appear in the web interface.
+                if api_event['EventAgendaStatusId'] == 1:
                     continue
-            
+
+                else:
+                    web_event = self.web_results(api_event)
```

Comment:
Should we raise an error in this instance instead? We probably want to know if we aren't finding events – could be that the key is inconsistent (as was the case with the NYC API events scraper, which resulted in no City Council meetings being scraped), or there are more conditions that result in an event in the API but not the web interface.

Reply:
That's not the current behavior. And I'm thinking about the case when we can't reliably know when an API event won't be in the web results.

Reply:
Cool to not changing current behavior. It would be nice to know when an event isn't in the web scrape, though. I'm thinking in particular about the City Council example – we would have had no way of knowing we were missing all City Council meetings until we noticed they weren't in the OCD API or Councilmatic. What do you think about logging something when that occurs?